Shared/Samples/Edit feature attachments/EditFeatureAttachmentsView.Model.swift

```diff
@@ -0,0 +1,112 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+extension EditFeatureAttachmentsView {
+    @MainActor
+    class Model: ObservableObject {
+        /// A map with ArcGIS Oceans basemap style.
+        let map = Map(basemapStyle: .arcGISOceans)
+        
+        /// The currently selected map feature.
+        var selectedFeature: ArcGISFeature?
+        
+        /// The placement of the callout on the map.
+        @Published var calloutPlacement: CalloutPlacement?
+        
+        /// Holds the attachments of the currently selected feature.
+        @Published private(set) var attachments: [Attachment] = []
+        
+        /// The text shown on the callout.
+        @Published private(set) var calloutText = ""
+        
+        /// The detail text shown on the callout.
+        @Published private(set) var calloutDetailText = ""
+        
+        /// The feature layer populated with data by the feature table using the remote service url.
+        let featureLayer = FeatureLayer(featureTable: ServiceFeatureTable(url: .featureServiceURL))
+        
+        init() {
+            map.addOperationalLayer(featureLayer)
+        }
+        
+        /// Selects the tapped feature.
+        /// - Parameter feature: The selected feature.
+        func selectFeature(_ feature: ArcGISFeature) async throws {
+            selectedFeature = feature
+            featureLayer.selectFeature(feature)
+            try await fetchAttachmentsAndUpdateFeature(feature: feature)
+        }
+        
+        /// Updates the callout text for the selected feature.
+        private func updateCalloutDetailsForSelectedFeature(feature: ArcGISFeature) {
```

Comment:
The Swift API Design Guideline has some examples on how to avoid repeating words. In this case, if I'm naming this method, I'll probably do one of the following:

- `updateCalloutDetails(for feature: ArcGISFeature)`
- `updateCalloutDetailsForSelectedFeature(_ feature: ArcGISFeature)`

Reply:
private func updateCalloutDetails(for feature: ArcGISFeature)